Makefile

```diff
@@ -1,23 +1,46 @@
-all: output/rootfs.cpio.xz output/bzImage
+TARGETS := output/rootfs.cpio.xz output/bzImage
+SOURCES := Dockerfile config/buildroot config/kernel config/user bin/build_dhyve
 
-output/rootfs.cpio.xz output/bzImage: | output
-	@docker inspect dhyve-os-built >/dev/null 2>&1 || ${MAKE} build
-	docker cp dhyve-os-built:/build/buildroot/output/images/$(@F) output/
+BUILD_IMAGE     := dhyve-os-builder
+BUILD_CONTAINER := dhyve-os-built
 
-build: | ccache dl
-	docker build --no-cache -t dhyve-os-build .
-	-docker rm dhyve-os-built
-	docker run -v ${PWD}/ccache:/build/buildroot/ccache -v ${PWD}/dl:/build/buildroot/dl --name=dhyve-os-built dhyve-os-build
+BUILT := `docker ps -aq -f name=$(BUILD_CONTAINER) -f exited=0`
+
+all: $(TARGETS)
+
+$(TARGETS): build | output
+	@if [ "$(BUILT)" != "" ]; then \
+		echo docker cp $(BUILD_CONTAINER):/build/buildroot/output/images/$(@F) output/; \
```

Comment:
Just I would like to see progresses.
They are unnecessary. You can remove them.